readme.md

```diff
@@ -156,24 +156,47 @@ An example would be an `Authorization` header with a signature depending on host
 While this could be calculated before making the request it is just convenient to have it done
 automatically with each request.
 
-For this purpose `Proc`s/`lambda`s can be set on the Grac object following the example below:
+For this purpose a class can be added as middleware which accepts at least one parameter during
+initialization and has a call method accepting the parameters as shown in the example below.
+The first parameter will always be the request object, i.e. the instance of `Grac` or another middleware
+already wrapper around it. Additional configuration can be provided to the middleware by accepting
+additional parameters. These will be passed along during the request when initializing the middleware.
+
 
 ```ruby
-mw = lambda do |opts, request_uri, method, params, body|
-  # your code here
-  # opts        - Hash of the options currently set on the grac object
-  # request_uri - uri returned by grac
-  # method      - http method (lower case)
-  # params      - hash of all params for this request
-  # body        - serialized body
-  return opts, request_uri, method, params, body
+class MW
+  def initialize(request, *params)
+    @request = request
+    @params  = params
```

Comment:
I think we shouldn't call the argument 'params' - might be confusing with query params below.